flow-data/src/main/java/com/vaadin/flow/data/binder/Binder.java

```
@@ -2332,13 +2333,86 @@ public void refreshFields() {
      *             if some of the bound field values fail to validate
      */
     public void writeBean(BEAN bean) throws ValidationException {
-        BinderValidationStatus<BEAN> status = doWriteIfValid(bean, bindings);
+        writeBean(bean, bindings);
+    }
+
+    /**
+     * Writes changes from the given bindings to the given bean if all
+     * validators (binding and bean level) pass.
+     * <p>
+     * If any field binding validator fails, no values are written and a
+     * {@code ValidationException} is thrown.
+     * <p>
+     * If all field level validators pass, the given bean is updated and bean
+     * level validators are run on the updated bean. If any bean level validator
+     * fails, the bean updates are reverted and a {@code ValidationException} is
+     * thrown.
+     *
+     * @see #writeBeanIfValid(Object)
+     * @see #writeBean(Object)
+     * @see #readBean(Object)
+     * @see #setBean(Object)
+     * @see #updateBean(Object)
+     *
+     * @param bean
+     *            the object to which to write the field values, not
+     *            {@code null}
+     * @param bindingsToWrite
+     *            Collection of bindings to use in writing the bean
+     * @throws ValidationException
+     *             if some of the bound field values fail to validate
+     */
+    public void writeBean(BEAN bean,
+            Collection<Binding<BEAN, ?>> bindingsToWrite)
```

Since we allow providing a collection of bindings from the outside, I wonder if we should add a sanity check to ensure that the given bindings belong to this Binder

---

If the collection contains _foreign_ bindings, we could either throw an exception or just log a warnin message and ignore them.

---

I prefer throwing IllegalArgumentException.